source/MapDetailPanel.cpp

```diff
@@ -322,6 +336,28 @@ bool MapDetailPanel::RClick(int x, int y)
 
 
 
+void MapDetailPanel::GeneratePlanetCards(const System &system)
+{
+	set<const Planet *> shown{};
+	
+	planetCards.clear();
+	MapPlanetCard::clear();
```

Comment:
I am not sure why I had to do this (clearing manually the list)
I mean I expected the vector clear to delete all elements within it given they are contained in it (that said for optimization this is probably best instead of having each element looking for itself in their private vector when they are deletef in this one)